Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISLoadAndProcess.py

```diff
@@ -0,0 +1,450 @@
+# -*- coding: utf-8 -*-
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from __future__ import (absolute_import, division, print_function)
+
+from mantid.api import (AlgorithmFactory, AnalysisDataService, DataProcessorAlgorithm,
+                        PropertyMode, WorkspaceGroup, WorkspaceProperty)
+
+from mantid.simpleapi import (AddSampleLog, LoadEventNexus, LoadNexus, Plus,
+                              RenameWorkspace)
+
+from mantid.kernel import (CompositeValidator, Direction, EnabledWhenProperty,
+                           IntBoundedValidator, Property, PropertyCriterion,
+                           StringArrayLengthValidator, StringArrayMandatoryValidator,
+                           StringArrayProperty)
+
+
+class Prop:
+    RUNS = 'InputRunList'
+    FIRST_TRANS_RUNS = 'FirstTransmissionRunList'
+    SECOND_TRANS_RUNS = 'SecondTransmissionRunList'
+    SLICE = 'SliceWorkspace'
+    NUMBER_OF_SLICES = 'NumberOfSlices'
+    OUTPUT_WS='OutputWorkspace'
+    OUTPUT_WS_BINNED='OutputWorkspaceBinned'
+    OUTPUT_WS_LAM='OutputWorkspaceWavelength'
+
+
+class ReflectometryISISLoadAndProcess(DataProcessorAlgorithm):
+
+    def __init__(self):
+        """Initialize an instance of the algorithm."""
+        DataProcessorAlgorithm.__init__(self)
+        self._tofPrefix = "TOF_"
+        self._transPrefix = "TRANS_"
+
+    def category(self):
+        """Return the categories of the algrithm."""
+        return 'ISIS\\Reflectometry;Workflow\\Reflectometry'
+
+    def name(self):
+        """Return the name of the algorithm."""
+        return 'ReflectometryISISLoadAndProcess'
+
+    def summary(self):
+        """Return a summary of the algorithm."""
+        return "Reduce ISIS reflectometry data, including optional loading and summing/slicing of the input runs."
+
+    def seeAlso(self):
+        """Return a list of related algorithm names."""
+        return ['ReflectometrySliceEventWorkspace', 'ReflectometryReductionOneAuto']
+
+    def version(self):
+        """Return the version of the algorithm."""
+        return 1
+
+    def PyInit(self):
+        """Initialize the input and output properties of the algorithm."""
+        mandatoryInputRuns = CompositeValidator()
+        mandatoryInputRuns.add(StringArrayMandatoryValidator())
+        lenValidator = StringArrayLengthValidator()
+        lenValidator.setLengthMin(1)
+        mandatoryInputRuns.add(lenValidator)
+        self.declareProperty(StringArrayProperty(Prop.RUNS,
```

Comment:
Could make use of `MultipleFileProperty`.

Here is code that we use:
```
 self.declareProperty(MultipleFileProperty(Prop.RUN,
                                                  action=FileAction.OptionalLoad,
                                                  extensions=['nxs']),
                      doc='A list of input run numbers/files.')
```

Reply:
I don't think I can use this because it also needs to support names for existing workspaces in the ADS